### ISSUE TITLE
Mismatched heads error when syncing multiple changes from different peers

### ISSUE THREAD

**Issue by acurrieclark** [automerge/automerge-repo#17]
**TLDR**: When syncing a large number of batched changes to another peer, a simultaneous update from another peer causes the document to error with a `error inflating document chunk ops: mismatching heads` error when saving then loading the document.

---

First a little context. I have been exploring how `automerge-repo` might work when used with an http based network adapter, rather than a tcp based one (eg. websocket). On the server end I am then storing the document in a postgres db using `Automerge.save`, then loading it back on each update using `Automerge.load`.

Having got it working, I thought it would be sensible to stress test it by creating a large number of changes to sync. This worked without issue, syncing successfully to an additional connected peer. However, due to the latency in the network and the load/save cycle, this took a short while to complete. My next test was to do the same thing, but simultaneously push a change from the additional peer. This is where I first saw the error.

To check that it wasn't an error with my http adapter implementation or improper transaction isolation with postgres, I mocked a version using (slightly modified) existing `automerge-repo` adapters. The result was the same. A reproduction is [available here](https://github.com/onsetsoftware/automerge-repo-mismatched-heads-example).

As this runs on one host, the only modification to the adapter was to introduce some latency so that the batched changes would take a short while to complete.

I suspect that there is a simpler way to recreate the issue, perhaps by simply trying to sync a change from a document which has not yet completed a round of sync messages to update it from another peer. 

---

@alexjg - My hunch is that this is an upstream `syncMessage` issue in `automerge` itself, but hopefully there is enough information here to help get to the bottom of it.

**Comment by acurrieclark** [comment]
This has been fixed in `automerge v2.0.2`
